server/consumer.go

```diff
@@ -48,7 +48,9 @@ type ConsumerInfo struct {
 }
 
 type ConsumerConfig struct {
+	// Durable is deprecated. All consumers will have names. picked by clients.
 	Durable         string          `json:"durable_name,omitempty"`
+	Name            string          `json:"name,omitempty"`
```

Comment:
Instead of adding a new property to check, why not simply just say all consumers are durable. And that durables can be reaped by the server based on their `inactive_threshold`. Server can still autoname "ephemerals" when name is empty.

This would simplify the logic on clients, at the expense that `durable_name` would always be set, and `name` in ConsumerInfo would duplicate it.

Reply:
The only other concern then becomes how to reflect the new use of the API. If the consumer specifies a filter in the config, and the server matches the version, the client could simply use the new API. Otherwise, it can use the old API, allowing security compliance of the `$JS.API.CREATE.CONSUMER...` API without having to deal with model changes.

Reply:
Durables could not be reaped, so I prefer to not change that behavior and leave as is for existing clients.

Going forward all consumers will have a name and optional inactive threshold.
Note also that consumer info already had Name, so it now simply matches that for new style.